homeassistant/core.py

```diff
@@ -1411,13 +1421,16 @@ def _async_fire(
                 "Bus:Handling %s", _event_repr(event_type, origin, event_data)
             )
 
-        listeners = self._listeners.get(event_type)
-        # EVENT_HOMEASSISTANT_CLOSE should not be sent to MATCH_ALL listeners
-        if event_type != EVENT_HOMEASSISTANT_CLOSE:
-            if listeners:
-                listeners = self._match_all_listeners + listeners
-            else:
-                listeners = self._match_all_listeners.copy()
```

Comment:
This was not correct, we didn't copy the list when the event was `EVENT_HOMEASSISTANT_CLOSE`